src/transformers/generation/utils.py

```diff
@@ -1147,6 +1147,21 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
         # `prepare_inputs_for_generation` doesn't accept them, then a stricter check can be made ;)
         if "kwargs" in model_args or "model_kwargs" in model_args:
             model_args |= set(inspect.signature(self.forward).parameters)
+
+        # Generic composite models need information from its 2 components
+        if self.__class__.__name__ in {
+            "EncoderDecoderModel",
+            "VisionEncoderDecoderModel",
+            "SpeechEncoderDecoderModel",
+        }:
+            # allow encoder kwargs
+            encoder_model_args = set(inspect.signature(self.encoder.forward).parameters)
+            model_args |= encoder_model_args
+
+            # allow decoder kwargs
+            decoder_model_args = set(inspect.signature(self.decoder.forward).parameters)
+            model_args |= {f"decoder_{x}" for x in decoder_model_args}
```

Comment:
```suggestion
```

The decoder inputs are prepared in `prepare_inputs_for_generation`, for which we were already doing the check. As such, these should not be needed :)

Reply:
A few lines above, we have

```python3
        model_args = set(inspect.signature(self.prepare_inputs_for_generation).parameters)
        ...
        if "kwargs" in model_args or "model_kwargs" in model_args:
            model_args |= set(inspect.signature(self.forward).parameters)
```
where we check `prepare_inputs_for_generation`, and we want to accept the arguments that are in the `forward` signature. This is True for both composite models as well as causal LM models.

But for encoder-decoder models, the model's `forward` don't have the full arguments in the decoder's `forward` signature. I believe we still need it. (otherwise, what's the reason to have the above block ..?)

Reply:
Uhmm I see. In that case, yeah, this can stay :)